mixpanel_oauth_to_redshift.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"crypto/md5"
+	"database/sql"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/Clever/pathio"
+	_ "github.com/lib/pq" // Postgres Driver
+)
+
+var (
+	aws_access_key_id     = os.Getenv("AWS_ACCESS_KEY_ID")
+	aws_secret_access_key = os.Getenv("AWS_SECRET_ACCESS_KEY")
+	aws_region            = os.Getenv("AWS_REGION")
+	mixpanel_api_key      = os.Getenv("MIXPANEL_API_KEY")
+	mixpanel_api_secret   = os.Getenv("MIXPANEL_API_SECRET")
+	redshift_host         = os.Getenv("AWS_REDSHIFT_HOST")
+	redshift_port         = os.Getenv("AWS_REDSHIFT_PORT")
+	redshift_database     = os.Getenv("AWS_REDSHIFT_DB_NAME")
+	redshift_user         = os.Getenv("AWS_REDSHIFT_USER")
+	redshift_password     = os.Getenv("AWS_REDSHIFT_PASSWORD")
+	jsonpaths_file        = flag.String("jsonpaths_file", "s3://clever-mixpanel/OAuth.json",
+		"s3 file with jsonpaths data.")
+	mixpanel_export_date = flag.String("mixpanel_export_date",
+		time.Now().AddDate(0, 0, -1).Format("2006-01-02"),
+		"Date in YYYY-MM-DD format. Defaults to yesterday.")
+	mixpanel_export_dir = flag.String("mixpanel_export_dir", "s3://clever-mixpanel/OAuth",
+		"Directory to store the exported mixpanel data.")
+	mixpanel_export_request_timeout = flag.Duration(
+		"mixpanel_export_request_timeout", 10*time.Minute,
+		"Timeout value to be used while making a mixpanel export request. Defaults to 10 mins.")
+	redshift_table       = flag.String("redshift_table", "mixpanel_oauth", "Name of the redshift table.")
+	export_from_mixpanel = flag.Bool("export_from_mixpanel", true, "Whether to export from mixpanel.")
+	copy_to_redshift     = flag.Bool("copy_to_redshift", true, "Whether to copy to redshift.")
```

Comment:
Changed all the golang names to mixedCaps in 7c05f5ac2c721fc5c4ea8dab5b1b73d0ea795d7c. The go tools seems to use all lowercase, no separator convention for its flags so I changed the flag names accordingly in 83dd471.